homeassistant/components/rainbird/binary_sensor.py

```diff
@@ -45,20 +56,10 @@ def is_on(self):
 
     def update(self):
         """Get the latest data and updates the states."""
-        _LOGGER.debug("Updating sensor: %s", self._name)
+        _LOGGER.debug("Updating sensor: %s", self._attr_name)
```

Comment:
```suggestion
        _LOGGER.debug("Updating sensor: %s", self.name)
```

Reply:
Just to understand it, in which cases do you prefer to use the property instead of the attribute? If I'm not mistaken, they should be equivalent here.

To name another example, take a look at the code from the [`nws`](https://github.com/home-assistant/core/pull/53293/files#diff-29b09cdc0f1e66e1fad19a50be145560be4f9a6b9fe02fd81cd08cdf77c231b6R87-R102) integration PR. Would you have preferred to `self.unit_of_measurement` there? (I can open a PR for it if that's the case.)

Reply:
CC: @frenck